core/lib/zksync_core/src/utils/mod.rs

```diff
@@ -262,4 +269,264 @@ mod tests {
+            unimplemented!("Not needed");
+        }
+
+        async fn nonce_at_for_account(
+            &self,
+            _: Address,
+            _: BlockNumber,
+            _: &'static str,
+        ) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn base_fee_history(
+            &self,
+            _: usize,
+            _: usize,
+            _: &'static str,
+        ) -> Result<Vec<u64>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn get_pending_block_base_fee_per_gas(
+            &self,
+            _: &'static str,
+        ) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn get_gas_price(&self, _: &'static str) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn block_number(&self, _: &'static str) -> Result<U64, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn failure_reason(&self, _: H256) -> Result<Option<FailureInfo>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn get_tx(
+            &self,
+            _: H256,
+            _: &'static str,
+        ) -> Result<Option<Transaction>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn tx_receipt(
+            &self,
+            _: H256,
+            _: &'static str,
+        ) -> Result<Option<TransactionReceipt>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn eth_balance(&self, _: H160, _: &'static str) -> Result<U256, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn call_contract_function(
+            &self,
+            _: ContractCall,
+        ) -> Result<Vec<ethabi::Token>, EthClientError> {
+            if self.tx_err {
+                return Err(EthClientError::EthereumGateway(
+                    zksync_types::web3::Error::Unreachable,
+                ));
+            }
+            if self.contract_err {
+                return Err(EthClientError::Contract(
+                    zksync_types::web3::contract::Error::InterfaceUnsupported,
+                ));
+            }
+            Ok(self.retval.clone())
+        }
+
+        async fn logs(&self, _: Filter, _: &'static str) -> Result<Vec<Log>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+
+        async fn block(
+            &self,
+            _: BlockId,
+            _: &'static str,
+        ) -> Result<Option<Block<H256>>, EthClientError> {
+            unimplemented!("Not needed");
+        }
+    }
+
+    #[tokio::test]
+    async fn ensure_l1_batch_commit_data_generation_mode_succeeds_when_both_match() {
+        let addr = Address::repeat_byte(0x01);
+        assert!(ensure_l1_batch_commit_data_generation_mode(
```

Comment:
Nit: here and in other places it'll probably be nicer if you'll use `assert_matches!` (it should already be in dev deps) to actually match on output.

Reply:
Sorry, I didn't check if we already had the dependency and didn't want to add it. I'll change it.

Reply:
OTOH, I'm not sure `anyhow` makes it easy to check which error was returned. I need to check that out first.

Reply:
I applied it for the `Ok(())` cases, but it seems unfeasible for error cases because the fields seem to be private.